analytics_data_api/v0/views/courses.py

```diff
@@ -39,19 +39,14 @@ def get(self, request, *args, **kwargs):
 
         return super(BaseCourseView, self).get(request, *args, **kwargs)
 
-    def verify_course_exists_or_404(self, course_id):
-        if self.model.objects.filter(course_id=course_id).exists():
-            return True
-
-        raise Http404
-
     def apply_date_filtering(self, queryset):
         raise NotImplementedError
 
     def get_queryset(self):
-        self.verify_course_exists_or_404(self.course_id)
         queryset = self.model.objects.filter(course_id=self.course_id)
         queryset = self.apply_date_filtering(queryset)
+        if not queryset:
```

Comment:
This won't work as expected. `queryset` will be truthy at this point because it has not yet been evaluated. I whipped up an alternative on the branch named 404-alternative. Feel free to use it, if you'd like.